### DISCUSSION TITLE
How about improved documentation or Wiki?

### DISCUSSION THREAD

**Discussion by hildebrau** [cryptk/haomnilogic-local#89] [General]
First, let me say THANK YOU.  The description sounds like just what I've been hunting for.  I'm not only a new OmniLogic pool owner, but also new to Home Assistant.  I find myself in the "drowning HA newbie" phase and still need to be hand fed, as I haven't learned all the tricks on my own, yet.  That said, I feel like your documentation is lacking for the newbies.  I get it.  I do the same when I write doc about stuff I know very well.  But, can we take a step back and try to write some doc/wiki/faq that's helpful for all levels of users, pretty please?

Granted, I haven't clicked on EVERY Issue to see if this was covered already... but I'll start at the basics.

Your README says:

> **Configuration**
> [..snip..]
> The only parameter you should need to configure is the IP address.
> 
> The Scan Interval setting controls how often the controller is polled. Per home-assistant recommendations/requirements, the minimum value is 5.

Great.  Sounds simple.  My first question is HOW/WHERE does one configure my IP address for this module to find it?  I am a network engineer, so configuring the IP on my router part (which I snipped) is no big deal for me.  Is it in HA's configuration.yaml?  Can you provide some sample config?

I figured this one out already; but I'd rewrite your Install section a little, too:

> **Installation**
> This repository is able to be installed via HACS, you will need to add this repo as a Custom Repository
> 

I'd suggest maybe:
**Installation**
This repository is able to be installed via HACS, you will need to add this repo as a Custom Repository using the following steps:
1. Open the HACS page/tab in Home Assistant
2. Navigate to the "Integrations" tab
3. Click on the three dots in the top right hand and select "Custom repositories" from the list of options
4. Copy and paste this repositories link https://github.com/cryptk/haomnilogic-local and paste it into the "Repository" input field
5. Select "Integration" from the "Category" dropdown
6. Click "Add"

Until I get past the configuration step, I won't have more suggestions.. but I haven't SEEN a lot of example configuration/ways to use this yet.. so if there is more to share from others, I'm all ears/eyes. ;)
Does this work with any other lights other than Colorlogic?  I have some bubbler lights which are NOT colorlogic.  They operate via a separate relay.  Separate from my other lights which ARE colorlogic.  I'd be interested in getting those supported eventually, if they aren't already.

Can we add a Helpers code wiki?  Short description of why we need the helper and what it does, what uses it, etc.

Thanks everyone!

**Comment by hildebrau** [comment]
> Your README says:
> 
> > **Configuration**
> > [..snip..]
> > The only parameter you should need to configure is the IP address.
> > The Scan Interval setting controls how often the controller is polled. Per home-assistant recommendations/requirements, the minimum value is 5.
> 
> Great. Sounds simple. My first question is HOW/WHERE does one configure my IP address for this module to find it? I am a network engineer, so configuring the IP on my router part (which I snipped) is no big deal for me. Is it in HA's configuration.yaml? Can you provide some sample config?
> 

I finally figured out what I was missing.  After adding the repo in HACS, I didn't go to the Settings->Integrations-> + ADD INTEGRATION and select **OmniLogic Local**.  THAT's where we enter the IP address.  LOL
<img width="412" alt="image" src="https://github.com/cryptk/haomnilogic-local/assets/2652477/4e30ba37-a46b-416f-9912-02825e0de695">

I still recommend updating the README to have a little more detail for us idiots. LOL

**Reply by kevinrossen** [reply]
Thank you for this!!! I could not figure out how to get it configured to work, but this did it for me!

**Reply by dan-riley** [reply]
7 months later, thank you, spent 20 minutes searching everywhere for what to do next!

**Comment by cryptk** [comment]
Making the installation documentation a little more clear is a great suggestion! I'll try and get that done at some point soon. 

Can you clarify what you mean regarding the helpers code? Are you talking about the riemann sum integral sensor helper? If so, I think that the sentence directly proceeding the steps to configure it explain what it's for. I would be open to any suggestions on how I could make that more clear though.

**Reply by kevinrossen** [reply]
I think for now all you need to do is add the steps mentioned above about adding the integration after getting the custom repo setup. I could not figure that out until I saw [hildebrau](https://github.com/hildebrau)'s comment.

**Reply by hildebrau** [reply]
> Can you clarify what you mean regarding the helpers code? Are you talking about the riemann sum integral sensor helper? If so, I think that the sentence directly proceeding the steps to configure it explain what it's for. I would be open to any suggestions on how I could make that more clear though.

I **think** maybe the "helpers" I was thinking about at the time were for the non-local omnilogic integration in order to get the lights to have a proper dashboard selection of colors/programs.  But hell, I could have the two projects still mixed up in my head, as documentation for both projects are pretty weak.. and my busy brain isn't helping any, either. ;)

I will say that the "local" omnilogic integration seems to be more solid than the cloud one.. that one seems to constantly need to be restarted (via restarting HA) for it to get data.. it just stops collecting state data randomly.. I've been slowly changing over my dashboards to use haomnilogic-local sensor data instead, now that I finally figured out how to finish the setup on this one.

As far as documentation, other than the updates previously mentioned, I would still find it helpful to see some examples of using this integration to monitor my spa/pool/bubbler lights to see if they've returned back to the default state.  For instance, if the pool panel loses power, I'd love for HA to help set them back to what they were prior, or to some predefined setting that we tend to like.    That's on my goal list, but I haven't had time to spend on it, yet.

Thanks again for the project!
